tensorboard/webapp/metrics/views/main_view/common_selectors_test.ts

```diff
@@ -430,4 +552,351 @@ describe('common selectors', () => {
+          new Map([
+            [
+              'lr',
+              {
+                type: DomainType.INTERVAL,
+                minValue: 0,
+                maxValue: 10,
+                filterLowerValue: 2,
+                filterUpperValue: 5,
+                includeUndefined: true,
+              },
+            ],
+          ]),
+          new Map()
+        ).length
+      ).toEqual(2);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map([
+            [
+              'who knows',
+              {
+                type: DomainType.INTERVAL,
+                minValue: 0,
+                maxValue: 10,
+                filterLowerValue: 2,
+                filterUpperValue: 5,
+                includeUndefined: false,
+              },
+            ],
+          ]),
+          new Map()
+        ).length
+      ).toEqual(0);
+    });
+
+    it('filters by metrics using interval filters', () => {
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map(),
+          new Map([
+            [
+              'foo',
+              {
+                type: DomainType.INTERVAL,
+                minValue: 0,
+                maxValue: 10,
+                filterLowerValue: 2,
+                filterUpperValue: 3,
+                includeUndefined: false,
+              },
+            ],
+          ])
+        ).length
+      ).toEqual(2);
+
+      expect(
+        selectors.TEST_ONLY.utils.filterRunItemsByHparamAndMetricFilter(
+          runTableItems,
+          new Map(),
+          new Map([
+            [
+              'bar',
+              {
+                type: DomainType.INTERVAL,
+                minValue: 0,
+                maxValue: 10,
+                filterLowerValue: 2,
+                filterUpperValue: 3,
+                includeUndefined: false,
+              },
+            ],
+          ])
+        ).length
+      ).toEqual(0);
+    });
+  });
+
+  describe('getFilteredRenderableRuns', () => {
+    it('does not use experiment alias when route is not compare', () => {
+      state.runs!.data.regexFilter = 'foo';
+      const result = selectors.getFilteredRenderableRuns(['exp1'])(state);
+      expect(result).toEqual([]);
+    });
+
+    it('uses experiment alias when route is compare', () => {
+      state.runs!.data.regexFilter = 'foo';
+      state.app_routing!.activeRoute!.routeKind = RouteKind.COMPARE_EXPERIMENT;
+      const result = selectors.getFilteredRenderableRuns(['exp1'])(state);
+      expect(result.length).toEqual(2);
+      expect(result[0].run.name).toEqual('run 1');
+      expect(result[1].run.name).toEqual('run 2');
+    });
+
+    it('filters runs by hparam and metrics', () => {
+      const spy = spyOn(
+        selectors.TEST_ONLY.utils,
+        'filterRunItemsByHparamAndMetricFilter'
+      ).and.callThrough();
```

Comment:
I think it might be nice to ensure it returns the result of the filtering. Maybe mock it to return one of the runs and then test that it is in the results? Also, ok with this if you feel it is not necessary.